File: c++/bogosort2.cpp
#include <algorithm>
#include <iostream>
#include <iterator>
#include <random>

using namespace std;

int main()
{
  int array[] = {3, 2, 4, 10, 1, 7, 9, 100};
  random_device rd;
  mt19937 rand(rd());
  while (!is_sorted(begin(array), end(array))) {
    shuffle(begin(array), end(array), rand);
  }
  for (const int n : array) {
    cout << n << " ";
  }
  return 0;
}

File: php/bogosort2.php
<?php
function isSorted($array)
{
  for ($i = 1; $i < count($array); $i++) {
    if ($array[$i-1] >= $array[$i]) {
      return FALSE;
    }
  }
  return TRUE;
}

$array = array(0, 3, 2, 4, 10, 1, 7, 9);
while (!isSorted($array)) {
  shuffle($array);
}
print_r($array);
